src/App.tsx

```diff
@@ -1,12 +1,65 @@
 import React from 'react';
-import './App.scss';
 
-// import { getAll, get5First, getRed } from './api/goods';
-// or
-// import * as goodsAPI from './api/goods';
+import { getAll, get5First, getRedGoods } from './api/goods';
+import { GoodsList } from './components/GoodsList';
 
-const App: React.FC = () => (
-  <h1>Dynamic list of Goods</h1>
-);
+interface State {
+  goods: Good[] | [];
+}
+
+class App extends React.Component<{}, State> {
+  state = {
+    goods: [],
+  };
+
+  componentDidMount() {
+    this.getGoods();
+  }
+
+  getGoods = () => {
+    getAll()
+      .then(goods => {
+        this.setState({ goods });
+      });
+  };
```

Comment:
you can create one similar method and get as an argument a needed function.
And try to use async/await